File: services/ats-system-checker/src/routes/health_routes.py
"""
Health Routes for ATS System
Health check and system status endpoints
"""

from fastapi import APIRouter, Depends
from typing import Dict, Any

from src.controllers import HealthController
from src.utils.dependencies import get_health_controller
from src.models.evaluation_models import APIResponse

router = APIRouter(tags=["health"])


@router.get("", summary="Health Check")
async def health_check(
    controller: HealthController = Depends(get_health_controller)
) -> APIResponse:
    """
    Health check endpoint - Returns system health status
    
    Returns:
    - Database connectivity status
    - AI services availability
    - Email service configuration
    - Overall system health
    """
    return await controller.health_check()


@router.get("/simple", summary="Simple Health Check")
async def simple_health_check() -> Dict[str, str]:
    """
    Simple health check endpoint
    
    Returns basic system status without dependencies.
    """
    return {
        "status": "healthy",
        "message": "ATS System is operational",
        "version": "2.0.0"
    }


@router.get("/test", summary="Test Health Check")
async def test_health_check() -> Dict[str, str]:
    """
    Test health check endpoint
    
    Returns test system status.
    """
    return {
        "status": "test",
        "message": "ATS System test endpoint",
        "version": "2.0.0"
    } 

File: services/ats-system-checker/src/routes/health.routes.ts
import { Router, Request, Response } from "express";
import { asyncHandler } from "../utils/asyncHandler";

const router = Router();

/**
 * @swagger
 * /ats-checker/health:
 *   get:
 *     summary: Health check endpoint
 *     description: Returns detailed health status of the ATS System
 *     tags: [Health]
 *     responses:
 *       200:
 *         description: System is operational
 *         content:
 *           application/json:
 *             schema:
 *               type: object
 *               properties:
 *                 success:
 *                   type: boolean
 *                   example: true
 *                 message:
 *                   type: string
 *                   example: "ATS System is operational"
 *                 data:
 *                   type: object
 *                   properties:
 *                     status:
 *                       type: string
 *                       example: "healthy"
 *                     version:
 *                       type: string
 *                       example: "2.0.0"
 *                     timestamp:
 *                       type: string
 *                       format: date-time
 */
router.get(
  "/health",
  asyncHandler(async (req: Request, res: Response) => {
    return res.status(200).json({
      success: true,
      message: "ATS System is operational",
      data: {
        status: "healthy",
        version: "2.0.0",
        timestamp: new Date().toISOString()
      }
    });
  })
);

/**
 * @swagger
 * /ats-checker/health/simple:
 *   get:
 *     summary: Simple health check endpoint
 *     description: Returns a simple health status response
 *     tags: [Health]
 *     responses:
 *       200:
 *         description: System is operational
 *         content:
 *           application/json:
 *             schema:
 *               type: object
 *               properties:
 *                 status:
 *                   type: string
 *                   example: "healthy"
 *                 message:
 *                   type: string
 *                   example: "ATS System is operational"
 *                 version:
 *                   type: string
 *                   example: "2.0.0"
 */
router.get(
  "/health/simple",
  asyncHandler(async (req: Request, res: Response) => {
    return res.status(200).json({
      status: "healthy",
      message: "ATS System is operational",
      version: "2.0.0"
    });
  })
);

export default router;

